autogpt/commands/file_operations.py

```diff
@@ -157,43 +161,42 @@ def read_file(filename: str) -> str:
         charset_match = charset_normalizer.from_path(filename).best()
         encoding = charset_match.encoding
         logger.debug(f"Read file '{filename}' with encoding '{encoding}'")
-        return str(charset_match)
+        content = str(charset_match)
+
+        # TODO: invalidate/update memory when file is edited
+        file_memory = MemoryItem.from_text_file(content, filename)
```

Comment:
Having the agent invoke a language model and store memory _implicitly_ every time it reads a file seems really not great.

Reply:
This we should fix, but fix later and involves broader scope than is in your work. I think we want to define a protocol for the commands so they execute actions and return results, and then the agent can decide what's important to summarize.

Ignore for now

Reply:
This needs further discussion imo, because I see this as a case of "not all LLM calls are created equal". Some are the agent's business, but in this specific case, they are just a tool to get summaries of data, not unlike how LangChain does extreme abstraction of functionality using LLMs.

I agree that it's probably a good idea to give the LLM a way to discriminate between content that should or should not be memorized, but that can wait until the memory actually works. As @p-i- keeps saying: first make it work, worry about cost optimization later.